src/index/run.go

```diff
@@ -48,11 +48,17 @@ func (idx Index) Run(
 		checksumUpToDate = realChecksum == stg.Checksum
 	}
 
-	// Run if we have a command and no inputs.
-	doRun := hasCommand && (len(stg.Inputs) == 0)
+	var doRun bool
 
-	// Run if our checksum is stale.
-	doRun = doRun || !checksumUpToDate
+	if hasCommand && (len(stg.Inputs) == 0) {
+		// Run if we have a command and no inputs.
+		doRun = true
+		logger.Info.Printf("running stage %s because there is a command and no inputs\n", stagePath)
```

Comment:
Could we have a `reason` string that sets a brief description, and then a single log line that prints the reason? I like enforcing the pattern of displaying `running stage STAGE_NAME (BRIEF DESCRIPTION)`.

This also highlights some missed opportunities to short circuit the "should run?" check. I think most of this function should be moved into a `checkRun` function (or similar name--haven't thought too hard about it). That way it can simply return to skip unnecessary checks. `checkRun` would return `doRun` and `reason`.

How do you feel about tackling this?